tests/unit/TodoList2.spec.ts

```diff
@@ -0,0 +1,230 @@
+//
```

Comment:
As you can see, I got almost nothing implemented here (design only)... and I'm not even happy with the design.

The Todo List component is brutally complicated due to all possible scenarios.

I would like to reorganize the Todo List but I am not sure how we should reorganize it: big code blocks for each possible state? But then there will be a lot of code/HTML duplication.

Clearly, the (partial) day I allocated to implement tests for all scenarios was not enough. I'm open to ideas on how to proceed.

Unfortunately there is no tangible benefit to our users to refactor this. There is only benefit to us in having easier/less fragile code to work on and potentially fewer bugs. But it's going to be a lot of work.

Thoughts?

Reply:
I could create a new refactor/cleanup ticket but I'm afraid we would never get around to implementing it.

Reply:
^^^ I'm going to create a new ticket for this anyways. I think it's our best option atm. (Fixing this now is not in scope.)

... As for this PR, do you think we should merge it?

Reply:
Yes to all of the above ( As in i 100% agree :) ). 

Like you said, best to stick with your scope for your sanity sake and we can tackle this when we're able.

Reply:
https://app.zenhub.com/workspaces/entity-5bf2f2164b5806bc2bf60531/issues/bcgov/entity/6168